packages/boxel-ui/test-app/tests/integration/components/resizable-panel-group-horizontal-test.gts

```diff
@@ -0,0 +1,310 @@
+    this.renderController.containerStyle =
+      'max-width: 100%; height: 200px; width: 218px;';
+    await renderHorizontalResizablePanelGroup(this.renderController);
+    this.renderController.containerStyle =
+      'max-width: 100%; height: 200px; width: 418px;';
+    await sleep(100); // let didResizeModifier run
+    assert.hasNumericStyle('.panel-0-content', 'width', 240, 1);
+    assert.hasNumericStyle('.panel-1-content', 'width', 160, 1);
+  });
+
+  test<MyTestContext>('it adjusts to its container growing (length specified)', async function (assert) {
+    this.renderController.containerStyle =
+      'max-width: 100%; height: 200px; width: 218px; border: 1px solid green';
+    this.renderController.panels[0].lengthPx = 100;
+    this.renderController.panels[1].lengthPx = 100;
+    await renderHorizontalResizablePanelGroup(this.renderController);
+    assert.hasNumericStyle('.panel-0-content', 'width', 100, 1.5);
+    assert.hasNumericStyle('.panel-1-content', 'width', 100, 1.5);
+    this.renderController.containerStyle =
+      'max-width: 100%; height: 200px; width: 418px; border: 1px solid green';
+    await sleep(100); // let didResizeModifier run
+    assert.hasNumericStyle('.panel-0-content', 'width', 200, 1.5);
+    assert.hasNumericStyle('.panel-1-content', 'width', 200, 1.5);
+  });
+
+  test<MyTestContext>('it adjusts to its container shrinking (default)', async function (assert) {
+    this.renderController.containerStyle =
+      'max-width: 100%; height: 200px; width: 418px;';
+    await renderHorizontalResizablePanelGroup(this.renderController);
+    this.renderController.containerStyle =
+      'max-width: 100%; height: 200px; width: 218px;';
+    await sleep(100); // let didResizeModifier run
+    assert.hasNumericStyle('.panel-0-content', 'width', 120, 1);
+    assert.hasNumericStyle('.panel-1-content', 'width', 80, 1);
+  });
+
+  test<MyTestContext>('it adjusts to its container shrinking (length specified A)', async function (assert) {
+    this.renderController.containerStyle =
+      'max-width: 100%; height: 200px; width: 420px; border: 1px solid green';
+    // Height ratio panel 1 and panel 2 is 3:2
+    this.renderController.panels[0].lengthPx = 240;
+    this.renderController.panels[1].lengthPx = 160;
+    await renderHorizontalResizablePanelGroup(this.renderController);
+    assert.hasNumericStyle('.panel-0-content', 'width', 240, 1);
+    assert.hasNumericStyle('.panel-1-content', 'width', 160, 1);
+    this.renderController.containerStyle =
+      'max-width: 100%; height: 200px; width: 220px; border: 1px solid green';
+    await sleep(100); // let didResizeModifier run
+    // Maintain the ratio 3:2 when resizing
+    assert.hasNumericStyle('.panel-0-content', 'width', 120, 1);
+    assert.hasNumericStyle('.panel-1-content', 'width', 80, 1);
+  });
+
+  test<MyTestContext>('it adjusts to its container shrinking (length specified B)', async function (assert) {
```

Comment:
length specified panel 2??

Reply:
It looked to me that A and B were actually about exercising that the ratio is maintained when the container shrinks, so I’ve removed one of them in cab6ad7.